ballista/core/src/physical_optimizer/task_group.rs

```diff
@@ -86,6 +88,7 @@ impl OptimizeTaskGroup {
                     Arc::new(CoalesceTasksExec::new(
                         node.clone().with_new_children(coalesce.children())?,
                         self.partitions.clone(),
+                        coalesce.output_ordering().map(|v| v.to_vec()),
```

Comment:
Hmm, I'm not sure this is valid.... 

If we move the coalesce up the sort expressions may not still be available (eg if there is a projection which eliminates a sort column).

I think it should always use it's direct child output ordering rather than the initial output ordering